タブの部品化(DAIKI)

i-k-i-a-d opened the pull request:
・タイムラインタブ・フォロー中タブ・フォロワータブを構成する3つのbladeファイルの共通部分をtemplate.blade.phpファイルにまとめた。
・3つのタブ用のbladeでtemplate.blade.phpファイルを読み込めるように実装。

レビューよろしくお願いします。

i-k-i-a-d committed:
タイムラインタブ・フォロー中タブ・フォロワータブを構成する3つのbladeファイルの共通部分をtemplate.blade.phpファイルにまとめた。
3つのタブ用のbladeでtemplate.blade.phpファイルを読み込めるように実装。

i-k-i-a-d committed:
Merge branch 'develop_b_fumiduki_rab' into feature/daiki/create_follow_template

i-k-i-a-d committed:
@sectionの第二引数で三項演算子の条件式を渡すのはシンプルに無駄な工程だと気付き、template.blade.phpファイル内に三つ分の三項演算子の条件式を記述。

Osk0830 commented on resources/views/layouts/template.blade.php:
```diff
@@ -0,0 +1,43 @@
+        <div class="m-auto" style="width: fit-content"></div>
+    </div>
+</div>
+@endsection
```
- `template.blade.php`という名称はあまりに汎用的すぎて、どのページに用いるテンプレートなのかが伝わらず、また今後の拡張の可能性などに鑑みても、再考してみた方がよくないでしょうか？

i-k-i-a-d replied:
改名しました。

Osk0830 commented on resources/views/layouts/template.blade.php:
```diff
@@ -0,0 +1,43 @@
+    </ul>
+         @yield('tab')
+        <div class="m-auto" style="width: fit-content"></div>
+    </div>
```
- この辺、インデントがおかしくなっているようです。再度調整してみてください。

i-k-i-a-d replied:
無駄なスペースがあったので削除しております。

Osk0830 commented on resources/views/layouts/template.blade.php:
```diff
@@ -0,0 +1,43 @@
+        <li class="nav-item"><a href="{{ route('users.followed', $user->id) }}" class="nav-link {{ Request::routeIs('users.followed') ? 'active' : '' }}">フォロワー</a></li>
+    </ul>
+         @yield('tab')
+        <div class="m-auto" style="width: fit-content"></div>
```
- この空 divはなんのためのものでしょうか？

i-k-i-a-d replied:
モックアップからコピペしただけなので元からあったようです。
確認したところ、コンテンツを中央ぞろえするためのdivのようです。ここでは不要と判断し削除しております。

Osk0830 commented on resources/views/users/followed.blade.php:
```diff
@@ -1,62 +1,23 @@
+            @endif
+        </li>
+    @endforeach
+</ul>
```
- この部分、ユーザーのリスト＋状況に応じてボタンが表示される、という点では、フォロータブもフォロワータブも同じではないでしょうか？ボタンだけ汎用的に使える形にしておけば、この部分を一つの部品として書き出してしまって、それぞれのBladeで`include`するような形にした方がすっきりしますし、メンテナンス性も向上すると思います。他の観点として、そもそもフォロー用のBladeとフォロワー用のBladeは別々にせず、同じものを用いる、という選択も可能かもしれないですね。

i-k-i-a-d replied:
フォロー用のbladeとフォロワー用のbladeを統合し、条件分岐で表示を切り替えれるように実装しました。

Osk0830 commented on resources/views/layouts/template.blade.php:
```diff
@@ -0,0 +1,43 @@
+                            @csrf
+                            <button type="submit" class="btn btn-primary btn-block">フォローする</button>
+                        </form>
+                    @endif
```
- ボタンだけ考えると、もしかしたらこの辺も、フォローページ・フォロワーページで使っているボタンと共通のものとして処理する可能性もあるかもしれませんね。もちろん、検討してみて共通部品化は困難、となるかもしれませんが、あくまでも可能性として書かせていただきました。

i-k-i-a-d replied:
検討しましたが、ボタンのaction属性のroute関数の第二引数でrouteに渡す値が異なるため共通化は困難と判断し、今回はボタンの部品化は見送りました。
また、最終期限が近く、実装したいコンテンツもまだまだたくさんあるため、あまりここに時間を使いたくないため今回はこれでよろしくお願いします。

Osk0830 reviewed (changes requested):
お疲れ様でした。
パーツ化のとっかかりはだいぶできてきているのではないかと思います。
部品化は、案件やルールによって、何をどう部品化するかが変わってくるものなので絶対解があるわけではないという点だけご理解いただいた上で、いくつか意見を書かせていただきましたので検討よろしくお願いします。

i-k-i-a-d committed:
template.blade.phpファイルをtab.blade.phpファイルへ改名。
followed.blade.phpファイルをfollow.blade.phpファイルへ改名。
follow.blade.phpファイルにフォロー中タブとフォロワータブの中身を条件分岐で表示切替できるように実装。
showFollowingメソッドとshowFollowedメソッドの戻り値で返すviewのファイル名をfollow.blade.phpへ修正。

Osk0830 commented on resources/views/layouts/tab.blade.php:
```diff
@@ -0,0 +1,42 @@
+                @endif
+            </div>
+        </div>
+    </aside>
+<div class="col-sm-8">
+    <ul class="nav nav-tabs nav-justified mb-3">
+        <li class="nav-item"><a href="{{ route('users.show', $user->id) }}" class="nav-link {{ Request::routeIs('users.show') ? 'active' : '' }}">タイムライン</a></li>
+        <li class="nav-item"><a href="{{ route('users.following', $user->id) }}" class="nav-link {{ Request::routeIs('users.following') ? 'active' : '' }}">フォロー中</a></li>
+        <li class="nav-item"><a href="{{ route('users.followed', $user->id) }}" class="nav-link {{ Request::routeIs('users.followed') ? 'active' : '' }}">フォロワー</a></li>
+    </ul>
```
- 前回お伝えしたこの部分のインデントの問題が解消されていないようです。

Osk0830 reviewed (approved):
お疲れ様でした！
こちら前回のレビューで書いたインデントの問題が残ってますが、
内容的にはOKだと思いますので、マージしますね！
